fw/http_msg.c

```diff
@@ -815,6 +820,33 @@ __hdr_sub(TfwHttpMsg *hm, const TfwStr *hdr, unsigned int hid)
 	return TFW_PASS;
 }
 
+/**
+ * Substitute request method with "get"
+ *
+ * Note: replaced method name length is supposed to be shorter
+ *       than the replacement string ("get")
+ */
+int
+tfw_http_req_meth_sub_with_get(TfwHttpMsg *hm, unsigned char dest_meth)
+{
+	const char *dest_name;
+	size_t dest_len;
+
+	TfwStr sub_str = TFW_STR_STRING("get");
+
+	BUG_ON(hm->req->method != dest_meth);
+
+	dest_name = tfw_http_tbl_method_get_name(dest_meth);
+	dest_len = strlen(dest_name);
+
+	BUG_ON(dest_len < sub_str.len);
+
+	if (ss_skb_cutoff_data(hm->msg.skb_head, &sub_str, 0, dest_len-sub_str.len))
```

Comment:
Just noticed that the comment from @krizhanovsky was lost after the update. That comment is comprehensive. This is not the right use of the SKB modification function. Also, the actual string replacement in the HTTP request (and the underlying SKB) seems to be missing.